core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -326,11 +328,14 @@ private void setResponse(BulkItemRequest request, BulkItemResponse response) {
         request.setPrimaryResponse(response);
         if (response.isFailed()) {
             request.setIgnoreOnReplica();
+        } else {
+            // Set the ShardInfo to 0 so we can safely send it to the replicas. We won't use it in the real response though.
+            response.getResponse().setShardInfo(new ShardInfo());
```

Comment:
👍  though the comment needs to be fix - it's not "so we can safely send it to replicas" but rather "safely serialize the response". We should probably mention that the true shard info is put on the ShardBulkResponse